### PR TITLE
Push all fixes related to issue #42

### PR DESCRIPTION
**Pull request by tfpk**
This is a "mega commit" which fixes many of the issues raised in #42.

Going to leave this here for further comment for a bit, and merge next week.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ macros in Rust. When completing each task, there are three goals:
 You should complete the kata in order, as they increase in
 difficulty, and depend on previous kata.
 
+This set of exericses is written for people who have already spent some time
```
Comment (s1z5d) [inline README.md]:
```suggestion
This set of exercises is written for people who have already spent some time
```

File: README.md [hunk 2]
```diff
@@ -10,6 +10,10 @@ macros in Rust. When completing each task, there are three goals:
...
+programming in Rust. Before completing this, work through a rust tutorial
```
Comment (s1z5d) [inline README.md]:
```suggestion
programming in Rust. Before completing this, work through a Rust tutorial
```

File: exercises/04_expression_variables/README.md [hunk 3]
```diff
@@ -19,8 +19,9 @@ It's also worth mentioning the fragment specifier `stmt`, which is similar to
 # Macros and the Precedence of Operators
 
 Macros do not affect the order of operations. If the expression `3 * math!(4,
-plus, 2)` expands to `3 * 4 + 2`, the answer will be 14, not 18 (as you might
-expect from the brackets).
+plus, 2)` expands to `3 * 4 + 2`, you might expect rust to interpret it as
```
Comment (s1z5d) [inline exercises/04_expression_variables/README.md]:
```suggestion
plus, 2)` expands to `3 * 4 + 2`, you might expect Rust to interpret it as
```

File: exercises/10_macros_calling_macros/README.md [hunk 4]
```diff
@@ -60,11 +60,13 @@ two token trees not separated by a `,` could not match).
...
 
 When forwarding a matched fragment to another macro-by-example, matchers in the
-second macro will see an opaque AST of the fragment type. The second macro can't
-use literal tokens to match the fragments in the matcher, only a fragment
-specifier of the same type. The `ident`, `lifetime`, and `tt` fragment types are an
-exception, and *can* be matched by literal tokens. The following illustrates this
-restriction:
+second macro will be passed an
+[AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree) of the fragment type,
+which cannot matched on except as a fragment of that type. The second macro
```
Comment (s1z5d) [inline exercises/10_macros_calling_macros/README.md]:
```suggestion
which cannot be matched on except as a fragment of that type. The second macro
```

File: exercises/12_hygienic_macros/README.md [hunk 5]
```diff
@@ -4,20 +4,19 @@ To quote [the reference](https://doc.rust-lang.org/reference/macros-by-example.h
 
 > By default, all identifiers referred to in a macro are expanded as-is, and are
 > looked up at the macro's invocation site. This can lead to issues if a macro
-> refers to an item or macro which isn't in scope at the invocation site. To
+> refers to an item (i.e. function/struct/enum/etc.) or macro which isn't in scope at the invocation site. To
```
Comment (s1z5d) [inline exercises/12_hygienic_macros/README.md:7]: nit: "scope at the invocation site. To" could be pushed to the next line etc.

**Review by s1z5d — COMMENTED** [review]
:) can't help with any of the rust, unfortunately.

File: exercises/12_hygienic_macros/README.md (hunk 5 above)
Comment (tfpk) [reply exercises/12_hygienic_macros/README.md:7]: Thanks, will fix separately.

**Pull request merged**